frontend/src/shared/schemas/validation/confirm-password-validator.ts

```diff
@@ -1,6 +1,7 @@
 import * as yup from 'yup';
 
-export const confirmPasswordValidator = yup
-  .string()
-  .oneOf([yup.ref('password'), null], 'Passwords must match')
-  .required('Confirm password is required');
+export const confirmPasswordValidator = (passwordRef: string = 'password') =>
+  yup
+    .string()
+    .oneOf([yup.ref(passwordRef), null], 'Passwords must match')
+    .required('Confirm password is required');
```

Comment:
i extended this validator so that we can choose the reference key to its input. by default, with no argument passed, the reference key will be `password`